### ISSUE TITLE
.csv table not exported after tracking

### ISSUE THREAD

**Issue by AnnaGallis** [ilastik/ilastik#1170]
Hello again and happy new year to everybody!

I wanted to export the spreadsheet with the values after tracking (using Ilastik 1.1.7, automatic workflow, pixel prediction map) using the Export menu in the Tracking applet, "Export tracking information" and choosing .csv, but it's not responding. When I click OK, nothing happens.

I have tried restarting the software, the computer and redoing the tracking from scratch creating a new project. Nothing useful.

Is there anything I'm doing wrong?
Thanks for your help,

AG

**Comment by chaubold** [comment]
Hi Anna,

No, you did everything correctly -- the User Interface is just not very intuitive yet... The CSV export behaves as the other export, in the settings dialog you only specify what should be exported and how. To export the real thing, you have to export images and CSV table together by clicking export on the dataset (or Export All). This inconvenience will hopefully be changed very soon...

Best,
Carsten

**Comment by AnnaGallis** [comment]
Hello Carsten,
thank you, as always, for answering so fast and your patience. Your suggestion solved partially the problem and at least I can keep working.

However, I said partially because recently I am having troubles exporting things, for instance:
- If I export the segmentation results generated in the Pixel Classification workflow (.h5) and then try to load them in the Automatic Tracking Workflow, the software won't recognize the map and I cannot load it. My workaround is to copy the .hd5 file generated in another folder, then it appears as an option when you click "Add image" in the Prediction Maps tab. 
- When I try to export the table with the tracking results (with the way you told me to do in the previous post), only the basic features are exported, as if I hadn't chosen all of the features in "Configure Table Export for Tracking+Features". I would really love to have the other features as well :)
- And now, I just got this error message (see attached file), trying to export a new tracking result table the way you told me. This time I cannot even get the simple table.

![exporting problem](https://cloud.githubusercontent.com/assets/14805382/12236699/9758ab9e-b87b-11e5-8739-326103725f54.jpg)

Is there something I can do?
Thanks a lot and all the best,

AG

**Comment by AnnaGallis** [comment]
Hello everybody, hello Carsten,
just an update on this issue, now I always get this error message I attached in the previous post when I try to export results, doesn't matter what. I proceeded like this: I went to the "Tracking Result Export" applet, choose "Tracking Result" as source and then choose .tiff in the "Choose Export Image Settings..." button. Then, I selected all the features for the .csv table in the "Configure Table Export for Tracking+Features". Finally, I clicked on the "Export All" button. Then I get the message above. Furthermore, if I go again to the "Configure Table Export for Tracking+Features" button, all my settings are reset to default.

As workaround, I can get the image sequence if I right click on "Selected Output - Preview" once I have checked "Tracking Result" as source and then click on "Export". But there is no way I can get the table with the results. This is really bad because without the table I cannot get my analysis done.

Is there something I'm doing wrong? Is there another workaround to get the table with all the tracking results and features?

Thank you very much for your attention.
All the best,

AG

**Comment by chaubold** [comment]
Putting this here so everyone can see the problems:

I wrote in an email:
>Sorry that I didn't reply to this earlier. I cannot really see what's going on, especially not where this error message is coming from. 
Could you send me your .ilastik_log.txt file that should be in your user account's home folder (e.g. C:\Documents and Settings\yourusername\.ilastik_log.txt)?
>Also, did you try using ilastik 1.1.8 instead of 1.1.7?

AG wrote:
> I had to wait till this morning to get permission to install the new version. I have tried opening my project in Ilastik 1.1.8 as you suggested, but it cannot be loaded. Attached you find the error I get and the .ilastik_log.txt file when I try to open the project using this version.
I have tried to reproduce the error I got while trying to export the table using Ilastik 1.1.7, to send you the .ilastik_log.txt file that you asked me for. However, since I installed 1.1.8 in the computer, Ilastik 1.1.7 is not even opening anymore. I cannot send you the log at the moment.

> Alternatively, I could try to redo the whole project in Ilastik 1.1.8 (which seems I will have to do anyway), and see if I can export the table then. I can contact you again whatever the result is, but it will take a while because I have to train all the classifiers from scratch. Anyway, it's a pity that you cannot come back to old projects anymore because the files are not compatible with the new versions. Maybe I am just doing something wrong.

And here is the excerpt from the ilastik log.

```
ERROR 2016-01-26 13:31:12,842 excepthooks 5676 5452 Unhandled exception in thread: 'TileProvider-5452'
ERROR 2016-01-26 13:31:12,842 excepthooks 5676 2584 Traceback (most recent call last):
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\volumina\volumina\tiling.py", line 90, in _render
    img = self.image_req.wait()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\volumina\volumina\pixelpipeline\imagesources.py", line 149, in wait
    return self.toImage()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\volumina\volumina\pixelpipeline\imagesources.py", line 155, in toImage
    self._arrayreq.wait()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\volumina\volumina\pixelpipeline\slicesources.py", line 43, in wait
    return self._sp(self._ar.wait())
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\volumina\volumina\pixelpipeline\datasources.py", line 471, in wait
    rawData = self._rawRequest.wait()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\volumina\volumina\pixelpipeline\datasources.py", line 205, in wrapper
    return func(*args, **kwargs)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\volumina\volumina\pixelpipeline\datasources.py", line 229, in wait
    a = self._req.wait()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 449, in wait
    return self._wait(timeout)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 472, in _wait
    self._wait_within_foreign_thread( timeout )
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 293, in _execute
    self._result = self.fn()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\slot.py", line 846, in __call__
    result_op = self.operator.execute(self.slot, (), self.roi, destination)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\operators\opReorderAxes.py", line 128, in execute
    self.Input( *in_roi ).writeInto( result_input_view ).wait()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 449, in wait
    return self._wait(timeout)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 472, in _wait
    self._wait_within_foreign_thread( timeout )
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 293, in _execute
    self._result = self.fn()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 736, in __call__
    return self.func( *totalargs, **self.kwargs)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\slot.py", line 846, in __call__
    result_op = self.operator.execute(self.slot, (), self.roi, destination)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\operators\opReorderAxes.py", line 128, in execute
    self.Input( *in_roi ).writeInto( result_input_view ).wait()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 449, in wait
    return self._wait(timeout)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 472, in _wait
    self._wait_within_foreign_thread( timeout )
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 293, in _execute
    self._result = self.fn()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 736, in __call__
    return self.func( *totalargs, **self.kwargs)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\slot.py", line 846, in __call__
    result_op = self.operator.execute(self.slot, (), self.roi, destination)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\operators\opReorderAxes.py", line 128, in execute
    self.Input( *in_roi ).writeInto( result_input_view ).wait()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 449, in wait
    return self._wait(timeout)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 472, in _wait
    self._wait_within_foreign_thread( timeout )
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 293, in _execute
    self._result = self.fn()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 736, in __call__
    return self.func( *totalargs, **self.kwargs)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\slot.py", line 846, in __call__
    result_op = self.operator.execute(self.slot, (), self.roi, destination)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\operators\valueProviders.py", line 120, in execute
    req.wait()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 449, in wait
    return self._wait(timeout)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 472, in _wait
    self._wait_within_foreign_thread( timeout )
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 293, in _execute
    self._result = self.fn()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 736, in __call__
    return self.func( *totalargs, **self.kwargs)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\slot.py", line 846, in __call__
    result_op = self.operator.execute(self.slot, (), self.roi, destination)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\operators\valueProviders.py", line 120, in execute
    req.wait()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 449, in wait
    return self._wait(timeout)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 472, in _wait
    self._wait_within_foreign_thread( timeout )
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 293, in _execute
    self._result = self.fn()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 736, in __call__
    return self.func( *totalargs, **self.kwargs)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\slot.py", line 846, in __call__
    result_op = self.operator.execute(self.slot, (), self.roi, destination)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\operators\opSplitRequestsBlockwise.py", line 100, in execute
    pool.wait()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 1148, in wait
    self._clear_finishing_requests()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 1229, in _clear_finishing_requests
    req.block()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 455, in block
    self._wait(timeout) # No return value. Use wait()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 472, in _wait
    self._wait_within_foreign_thread( timeout )
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 293, in _execute
    self._result = self.fn()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\slot.py", line 846, in __call__
    result_op = self.operator.execute(self.slot, (), self.roi, destination)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\operators\opUnblockedArrayCache.py", line 100, in execute
    self.Input(roi.start, roi.stop).writeInto(result).block()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 455, in block
    self._wait(timeout) # No return value. Use wait()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 475, in _wait
    self._wait_within_request( current_request )
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 293, in _execute
    self._result = self.fn()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 736, in __call__
    return self.func( *totalargs, **self.kwargs)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\slot.py", line 846, in __call__
    result_op = self.operator.execute(self.slot, (), self.roi, destination)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\operators\opCacheFixer.py", line 63, in execute
    self.Input(roi.start, roi.stop).writeInto(result).wait()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 449, in wait
    return self._wait(timeout)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 475, in _wait
    self._wait_within_request( current_request )
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 293, in _execute
    self._result = self.fn()
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\request\request.py", line 736, in __call__
    return self.func( *totalargs, **self.kwargs)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\slot.py", line 846, in __call__
    result_op = self.operator.execute(self.slot, (), self.roi, destination)
  File "C:\Program Files\ilastik-1.1.8\ilastik-meta\lazyflow\lazyflow\operators\ioOperators\opTiffReader.py", line 139, in execute
    page_data = vigra.impex.readImage(self._filepath, dtype='NATIVE', index=tiff_page_list_index, order='C')
ArgumentError: Python argument types in
    vigra.impex.readImage(str)
did not match C++ signature:
    readImage(char const * __ptr64 filename, class boost::python::api::object dtype='FLOAT', unsigned int index=0, class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > order='')
```

@stuarteberg @ukoethe: Did we introduce an incompatibility while loading project files due to the changed tiff reader?

@AnnaGallis if we're not able to reopen/convert your old project files, we should at least be able to transfer your labels for training the classifiers so that you don't have to train everything again!

**Comment by stuarteberg** [comment]
>@stuarteberg @ukoethe: Did we introduce an incompatibility while loading project files due to the changed tiff reader?

This is not an error related to the project file format, it's a problem with opening tiff files.  It's a duplicate of #1176 and #1178.  It will be fixed as soon as we can make a new release.

**Comment by AnnaGallis** [comment]
Hello everyone!
Just a quick update (I think it may even close the issue). I tried to reproduce the error I got while exporting (the one with the screenshot above) in another computer, since now that I have installed Ilastik 1.1.8 in mine I cannot open Ilastik 1.1.7 anymore.
To my surprise, I could export everything, no problem!
I guess that it's a problem im my computer messing up with the software. I am going to uninstall and re-install everything and see if it works again. Otherwise, I will just use another computer for exporting my data while the issue with the opening tiff files in Ilastik 1.1.8 is solved :)

Thank you again, as always, for your help and your patience!

AG
